### PR TITLE
Add support for IPv6 with ip6tables

### PR DESCRIPTION
**Pull request by robbertkl**
For IPv6 filtering, `ip6tables` is used instead of `iptables`. Since the command line arguments are the same, this library can easily be used to control both of them.

This PR adds support for using it with `ip6tables` by instantiating it with `NewWithProtocol(ProtocolIPv6)`. For backwards compatibility, `New()` still creates a good old IPv4 version.

All tests run on the IPv6 version as well, but use IPv6 addresses instead of IPv4 addresses. I've also changed the IPv4 addresses used in the tests to non-existing addresses meant for documentation (see [RFC 5737](https://tools.ietf.org/html/rfc5737))

### CODE REVIEW

**Comment by vincentbernat** [comment]
Any chance this gets merged?

**Comment by jonboulle** [comment]
@steveeJ @squeed review?

File: iptables/iptables.go [hunk 1]
```diff
@@ -39,23 +39,41 @@ func (e *Error) Error() string {
...
+)
+
 type IPTables struct {
 	path     string
+	proto    Protocol
 	hasCheck bool
 	hasWait  bool
 }
 
+// Creates a new IPTables.
```
Comment (lucab) [inline iptables/iptables.go]: Here and further below, please follow the complete-sentence suggestion from https://blog.golang.org/godoc-documenting-go-code, e.g. `New creates a...`.

File: iptables/iptables_test.go [hunk 2]
```diff
@@ -119,7 +123,20 @@ func TestRules(t *testing.T) {
...
+	var address1, address2, subnet1, subnet2 string
+	if ipt.proto == ProtocolIPv6 {
+		address1 = "2001:db8::1/128"
+		address2 = "2001:db8::2/128"
+		subnet1 = "fd00:a::/48"
+		subnet2 = "fd00:b::/48"
+	} else {
+		address1 = "203.0.113.1/32"
+		address2 = "203.0.113.2/32"
+		subnet1 = "10.1.0.0/16"
```
Comment (lucab) [inline iptables/iptables_test.go]: You mentioned changing to RFC5737, any reasons for keeping these (and the v6 above) from RFC1918 instead? Is iptables unhappy with that?

**Comment by lucab** [comment]
I guess it may be worthy exposing a `Iptables.Proto()` now, so that consumers may case-switch on it when passing reference around.

**Comment by robbertkl** [comment]
Hi @lucab, thanks for your review!

I've added 2 commits fixing the documentation style issue and adding the `Iptables.Proto()` you've suggested, including some tests for it. Hope this is what you've had in mind.

As for you comment on the RFC5737 change, I've actually left the RFC1918 addresses intact: `subnet1` and `subnet2` are `10.1.0.0/16` and `10.2.0.0/16` respectively, just like before. However, the tests were also using `8.8.8.8/32` and `9.9.9.9/32`, which are in fact real, live addresses (you might recognize `8.8.8.8` as 1 of Google's public DNS resolvers). While this won't do any damage when running the tests, I found it to be "cleaner" to use non-existent addresses, like the `address1` and `address2` I introduced from RFC5737. Does that clear it up?

**Comment by lucab** [comment]
Sorry if I was not clear, my question was the opposite: why retaining the private subnets instead of updating to RFC5737 everywhere?

**Comment by robbertkl** [comment]
Fair point, changed the subnets to RFC5737 as well.

Is it good to go like this? Do you want me to squash everything into 1 commit? @squeed

**Comment by lucab** [comment]
@robbertkl Thanks! Somebody more familiar than me with this codebase will have a final pass at this soon. Commits will need to be squashed, yes.

**Comment by squeed** [comment]
This looks good to me. Please do squash the commits.

**Comment by robbertkl** [comment]
Done

**Comment by lucab** [comment]
LGTM. Thanks @robbertkl!

**Pull request merged**